py4cast/datasets/smeagol.py

```diff
@@ -431,6 +446,43 @@ def __len__(self):
             raise ValueError("No valid sample in the dataset.")
         return length
 
+    def generate_toa_radiation_forcing(self, sample: Sample):
```

Comment:
This function could be useful and desired with any dataset, have you considered moving it to datasets/base.py ?

Reply:
ALso this kind of very calculatory isolated functions are ideal candidate for unit tests ! consider adding one ?

Reply:
The function could be decoupled from the Sample object by taking directly two 'simple' arguments: day_of_year and hour_of_day.

Dataset maintainers could then choose to use it or not, and how they plug it with their sample implementation.

@CorentinSeznec for unit test example: https://github.com/meteofrance/py4cast/blob/main/tests/test_datasets.py